setup/backup.class.inc.php

```diff
@@ -486,7 +486,14 @@ public static function GetMysqlCliTlsOptions($oConfig)
 		$sMysqlSSLModeVersion = '5.7.0'; //Mysql 5.7.0 and upper deprecated --ssl and uses --ssl-mode instead
 		if ($sDBVendor === CMDBSource::ENUM_DB_VENDOR_MYSQL && version_compare($sDBVersion, $sMysqlSSLModeVersion, '>='))
```

Comment:
Also, since iTop 3.x only supports MySQL 5.7+ the version check can be removed and this part should always be executed?

Reply:
Oh wait, does MariaDB still use `--ssl`? In that case, it can be this:
```suggestion
		if (CMDBSource::GetDBVendor() === CMDBSource::ENUM_DB_VENDOR_MYSQL)
```

Reply:
Another note on this: It's perfectly possible to have a DB server with MySQL and the `mysqldump` tool from MariaDB and vice versa, so be careful with these checks.

Reply:
> Also, since iTop 3.x only [supports](https://www.itophub.io/wiki/page?id=3_0_0:install:requirements) MySQL 5.7+ the version check can be removed and this part should always be executed?

Depends if we manage to make this merged in the `support/2.7` branch 🤞

Reply:
Currently it is targeted at `develop`, which means iTop 3.1
My other point still counts..

Reply:
> Currently it is targeted at develop, which means iTop 3.1

> Also this is a good candidate to be merged in the support/2.7 branch, but we'll have to wait the functional review to discuss this.

Reply:
Correct, but my other point still stands..

Reply:
Damn you are right @Hipska `CMDBSource::GetDBVendor` will check the server version / vendor not the client, missed that :-(

I'll try to find a way to check the client version and come back as soon as I got a solution..

Reply:
OK I checked it.
Looks like the only other way to figure out the client version, would be by executing `mysqldump --version`
So it might be possible to do something like this:
```
$sMySQLDump = $this->GetMysqldumpCommand() . " --version";
exec($sCommand, $aOutput, $iRetCode);
```
and use a RegEx to get the vendor and the version from the output (not sure how is is such a RegEx since the version string ist not really standardized).

I am really not sure how "stable" such an implementation is...
It does feel a little bit "dirty" to grep for the version in the output of an exec command, doesn't it?

Anyhow the change would be bigger that originally planned from my side..

Another way would be the add a warning in the documentation, that it would be recommended to use that same mysql client and server version in case of an TLS encryption (in fact this warning is true for the current code without this PR as well, since currently you are already using the server version to check which client options to be set). But for sure a comment in the documentation will not address the real technical issue..

What do you think @Molkobain @Hipska ?

Reply:
IMO as said before, only a check if it is MySQL or MariaDB should be enough. (then it is not possible to backport to 2.7)

Reply:
It that case the problem regarding the usage of DB server with MySQL and the mysqldump tool from MariaDB (and vice versa) will still exist an a brief note in the documentation would make sense anyhow, wouldn't it?

Reading to backport or not, I do not have a strong opinion, so up to the Combodo as the maintainer :-)

Reply:
To me it depends how much we want to backport this fix to the `support/2.7`.

It would be great to do it, but if we are not sure on how to detect the vendor (uncertain RegEx), then I'll rather just fix this in the `support/3.0` to ensure stability.
Also as this is the first time this issue is reported, it might not be some frequent, so the need to backport it is not that big to me.

So long story short I'll go the Hipska way.

Reply:
Fine for me, I've already updated the code ✅